repos/system_upgrade/el8toel9/actors/removeupgradeefientry/tests/test_removeupgradeefientry.py

```diff
@@ -0,0 +1,103 @@
+
+    result = removeupgradeefientry.get_workaround_efi_info()
+    assert result == TEST_EFI_INFO
+
+
+def test_get_workaround_efi_info_multiple_entries(monkeypatch):
+    logger = logger_mocked()
+    monkeypatch.setattr(api, 'current_actor', CurrentActorMocked(
+        msgs=[TEST_EFI_INFO, TEST_EFI_INFO]))
+    monkeypatch.setattr(api, 'current_logger', logger)
+
+    result = removeupgradeefientry.get_workaround_efi_info()
+    assert result == TEST_EFI_INFO
+    assert 'Unexpectedly received more than one UpgradeEFIBootEntry message.' in logger.warnmsg
+
+
+def test_get_workaround_efi_info_no_entry(monkeypatch):
+    monkeypatch.setattr(api, 'current_actor', CurrentActorMocked(msgs=[]))
+
+    with pytest.raises(StopActorExecutionError, match='Could not remove UEFI boot entry for the upgrade initramfs'):
+        removeupgradeefientry.get_workaround_efi_info()
+
+
+def test_copy_grub_files(monkeypatch):
+    copy_file_calls = []
+
+    def mock_copy_file(src, dst):
+        copy_file_calls.append((src, dst))
+
+    monkeypatch.setattr(removeupgradeefientry, '_copy_file', mock_copy_file)
+    monkeypatch.setattr(os.path, 'exists', lambda path: True)
+
+    removeupgradeefientry._copy_grub_files(['required'], ['optional'])
+
+    assert (
+        os.path.join(removeupgradeefientry.LEAPP_EFIDIR_CANONICAL_PATH, 'required'),
+        os.path.join(removeupgradeefientry.RHEL_EFIDIR_CANONICAL_PATH, 'required'),
+    ) in copy_file_calls
+    assert (
+        os.path.join(removeupgradeefientry.LEAPP_EFIDIR_CANONICAL_PATH, 'optional'),
+        os.path.join(removeupgradeefientry.RHEL_EFIDIR_CANONICAL_PATH, 'optional'),
+    ) in copy_file_calls
+
+
+def test_copy_grub_files_missing_required(monkeypatch):
+    monkeypatch.setattr(os.path, 'exists', lambda path: False)
+
+    with pytest.raises(StopActorExecutionError, match='Required file required does not exists'):
+        removeupgradeefientry._copy_grub_files(['required'], [])
+
+
+def test_remove_upgrade_efi_entry(monkeypatch):
+    run_calls = []
+    copy_grub_files_calls = []
+    monkeypatch.setattr(api, 'current_actor', CurrentActorMocked(msgs=[TEST_EFI_INFO]))
+
+    def mock_run(command, checked=False):
+        run_calls.append(command)
+        return {'exit_code': 0}
+
+    def mock_copy_grub_files(required, optional):
+        copy_grub_files_calls.append((required, optional))
+
+    monkeypatch.setattr(removeupgradeefientry, '_copy_grub_files', mock_copy_grub_files)
+    monkeypatch.setattr(removeupgradeefientry, 'run', mock_run)
+
+    removeupgradeefientry.remove_upgrade_efi_entry()
+
+    assert ['/bin/mount', '/boot'] in run_calls
+    assert ['/bin/mount', '/boot/efi'] in run_calls
+    assert ['/usr/sbin/efibootmgr', '--delete-bootnum', '--bootnum', '0002'] in run_calls
+    assert ['rm', '-rf', removeupgradeefientry.LEAPP_EFIDIR_CANONICAL_PATH] in run_calls
+    assert ['/usr/sbin/efibootmgr', '--bootnext', '0001'] in run_calls
+    assert ['/bin/mount', '-a'] in run_calls
```

Comment:
The order of these calls is important, I would suggest popping the lists from `run_calls` when comparing them to also test that.

Reply:
Good idea, thanks! Changed.